pkg/controller/oneagent/oneagent_controller.go

```diff
@@ -273,116 +282,53 @@ func (r *ReconcileOneAgent) reconcileRollout(logger logr.Logger, instance dynatr
 	}
 
 	if instance.GetOneAgentStatus().Version == "" {
-		desired, err := dtc.GetLatestAgentVersion(dtclient.OsUnix, dtclient.InstallerTypeDefault)
-		if err != nil {
-			return false, fmt.Errorf("failed to get desired version: %w", err)
+		if instance.GetOneAgentSpec().UseImmutableImage {
+			if instance.GetOneAgentSpec().AgentVersion == "" {
+				instance.GetOneAgentStatus().Version = "latest"
```

Comment:
You can get the latest from `GetLatestAgentVersion()`

Reply:
Changed